src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1068,6 +1062,19 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
             # and it matches the spec.
             port = spec.get_port()
 
+        if spec.generate_cert:
```

Comment:
Just to make sure this second check has not been omitted by error: `not spec.rgw_frontend_ssl_certificate`

Reply:
we only needed that before because it was generating a service wide cert and setting it as the `rgw_frontend_ssl_certificate` field for the spec. Having `not spec.rgw_frontend_ssl_certificate` meant we wouldn't continually regenerate that cert when there was no reason to. Now we're generating them per daemon and aren't attaching them to the spec so there's no reason to check that field. Having `generate_cert: True` overrides having a frontend cert set in the spec.

Reply:
I added a validation check to the rgw service spec to make this more clear